src/Microsoft.Identity.Client/AppConfig/AuthorityInfo.cs

```diff
@@ -81,7 +81,7 @@ internal static AuthorityInfo FromAuthorityUri(string authorityUri, bool validat
         internal static AuthorityInfo FromAadAuthority(Uri cloudInstanceUri, Guid tenantId, bool validateAuthority)
         {
 #pragma warning disable CA1305 // Specify IFormatProvider
-            return FromAuthorityUri(string.Format(CultureInfo.InvariantCulture, "{0}/{1}/", cloudInstanceUri, tenantId.ToString("D")), validateAuthority);
+            return FromAuthorityUri(string.Format(CultureInfo.InvariantCulture, "{0}{1}/", cloudInstanceUri, tenantId.ToString("D")), validateAuthority);
```

Comment:
No need for additional slash here. According to the Uri specs,
 `"Note that the absolute path cannot be empty; if none is present in the original URI, it MUST be given as "/" (the server root)."` [Spec](https://tools.ietf.org/html/rfc2616#section-5.1.2)
Thus, there will always be a "/" appended to the uri so we dont need the additional one here. Otherwise you end up with "https://login.microsoftonline.com//tenantID"

Reply:
I don't understand the explanation, however if it means that the uri class will always return the uri including the trailing / that is great. Is that what the above means? 

---
In reply to: [306497215](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1281#discussion_r306497215) [](ancestors = 306497215)

Reply:
yes. if you create a uri using the uri class it will always append a trailing "/"